teacher/take_attendance.py

```diff
@@ -82,10 +82,17 @@ def RedirectToSelf(self, institution, session, selected_cid, selected_date):
 
   def post(self):
     auth = authorizer.Authorizer(self)
-    if not auth.HasTeacherAccess():
+    if not (auth.CanAdministerInstitutionFromUrl() or
+            auth.HasTeacherAccess()):
       auth.Redirect()
       return
 
+    user_type = 'None'
+    if auth.CanAdministerInstitutionFromUrl():
+      user_type = 'Admin'
+    elif auth.HasTeacherAccess():
+      user_type = 'Teacher'
```

Comment:
`user_type` is passed to the client side to display the correct menu (portal) - either admin or teacher.